packages/aws-cdk/lib/api/util/cloudformation/stack-status.ts

```diff
@@ -25,20 +25,16 @@ export class StackStatus {
     return this.name.endsWith('FAILED');
   }
 
-  get isRollback(): boolean {
-    return this.name.indexOf('ROLLBACK') !== -1;
-  }
-
-  get isStable(): boolean {
-    return !this.name.endsWith('_IN_PROGRESS');
+  get isInProgress(): boolean {
+    return this.name.endsWith('_IN_PROGRESS');
   }
 
   get isNotFound(): boolean {
     return this.name === 'NOT_FOUND';
   }
 
-  get isSuccess(): boolean {
-    return !this.isNotFound && !this.isRollback && !this.isFailure;
+  get isDeploySuccess(): boolean {
+    return !this.isNotFound && (this.name === 'CREATE_COMPLETE' || this.name === 'UPDATE_COMPLETE');
```

Comment:
What about `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS`? Will the stack move to `UPDATE_COMPLETE` after the clean up is complete?

Reply:
it *should* end up in UPDATE_COMPLETE but can also get stuck in this state (which cannot be updated).

In both of our usage of `waitFor*` APIs, we wait for the stack to stabilize before we evaluate it's status.
We return `undefined` if it's in an `IN_PROGRESS` state and the stack otherwise if they pass the conditional logic.

I do think you're right that we might need to be more graceful. I haven't used nested stacks a ton yet, but most of the info I could dig up on this status is largely in that domain.

We have this block of code in `deployStack`:

```ts
      const finalStack = await waitForStackDeploy(cfn, deployName);
      // This shouldn't really happen, but catch it anyway. You never know.
      if (!finalStack) { throw new Error('Stack deploy failed (the stack disappeared while we were deploying it)'); }
```

In the `UPDATE_ROLLBACK_IN_PROGRESS` scenario, we would return undefined as the 'stack' and that would throw an error stating

> Stack deploy failed (the stack disappeared while we were deploying it

That might be misleading although it is there with a comment of "this shouldn't really happen".

Reply:
Not sure I understand.. are you saying that we know of a case where the "this should not happen" actually happens?

Reply:
@eladb - not quite what I was trying to say. It's that the "this should not happen" block has a blanket error message that would be confusing if it ever does happen. I guess it's just general uneasiness of that code existing at all.

Back to `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS`, I've tested that if we end up there after a deploy, we will fail because we only deem an update to be successful if it ends up in `CREATE_COMPLETE` or `UPDATE_COMPLETE`. Otherwise, some work that we have performed did not succeed and we bail on deploying that stack and others it depended on.

Do you think something needs to change about how we check the deploy success status?

Reply:
If this is a possible status after a successful deploy, then it sounds like we need to consider it a successful deploy (am I missing something?)

Reply:
@eladb - it's *possible* as part of a successful deploy (could be on the path to success) but does not indicate success. It could also be a part of a deploy that gets stuck. The two possible outcomes when we see `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS` are:
* it transitions to `UPDATE_COMPLETE` after that, which means we succeeded. This is the only state it can transition to.
* It gets stuck and stays in current state - which means we failed. It cannot be updated after that point and will need to be deleted/cleaned up.

Essentially, this status alone is not indicative of how the deployment went.

Reply:
Okay. Let's revisit if needed